### PR TITLE
Event listeners Lesson

### PR DESCRIPTION
**Pull request by hfaerber**
# Curriculum PR Template

### Does this PR close any issues?
*This PR closes issue #90 *

### Description
* pulls in event listener lesson and uncomments link on SE2 lesson page
* revises learning goals and ensures codepen practices directly meet learning goals
* includes event and event.target in lesson since we cut the event bubbling lesson
* links to legacy FE event bubbling lesson as 'extra practice' resource (ive tried to avoid linking to FE/BE links but I think its valuable in this case)
* adds clearly outlined key takeaways
* removes inline styling practice and focuses practice exercises on styling via adding/removing classes and adding elements to the DOM dynamically

### What questions do you have/what do you want feedback on? (optional)
* Because I had to cover event listeners and using event.target all in one lesson, this lesson became a big one - is it too much? I tried to keep it directly pointed at the learning goals
  * if it is too much, we may need to move some into the Intro to DOM lesson.
* Please check for any spelling and syntax errors
* Please share feedback if formatting could be improved.

### CODE REVIEW

File: module2/lessons/js_event_listeners.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+Consider:  
+- Why don't we invoke the function within the eventListener?  What happens if we do?
+</section>
+
+## Explore
+
+There are many types of events beyond just "click".   Explore this list of [Mouse Events](https://developer.mozilla.org/en-US/docs/Web/API/Element#mouse_events) on MDN. Scroll beyond just mouse events to explore keyboard events, focus events, etc.  
+
+Open up a webpage that you often visit. Maybe it's a social media page, maybe it's your bank's website, maybe it's a news site. Take a few minutes and list all the event listeners that you think might be on that page! What do they do?
```
Comment (nicktu12) [inline module2/lessons/js_event_listeners.md:52]: Consider adding more guiding questions here, like "What are event listens that arent click events?" or "Does the display change on user interactions?" etc

File: module2/lessons/js_event_listeners.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+<section class="call-to-action">
+### Pair Challenge
+
+DOM manipulation based on user interaction requires putting several pieces together which is challenging! It is valuable to pseudocode out the steps of what you're trying to do before you try to do it in code.
+
+Fork [this Codepen](https://codepen.io/turingschool/pen/qBzNraG).
```
Comment (nicktu12) [inline module2/lessons/js_event_listeners.md:59]: We could consider adding a "Switch driver/navigator positions" halfway through this activity to give both students practice

File: module2/lessons/js_event_listeners.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+* What are 3 steps required when doing DOM manipulation based on user interaction?
+* How do you access the values a user has entered?
+* What is the preferred way to change styles programmatically?
+* How do you access the element that a click event (or other type of event) occurred on?
+* How do you access and manipulate elements that have been dynamically added to the DOM (since you can't use a simple querySelector)?
+
+</section>  
+
+## Suggested re-teaching practice
+
```
Comment (nicktu12) [inline module2/lessons/js_event_listeners.md:285]: Can we add a blog post or more links to different types of content here?

File: module2/lessons/js_event_listeners.md (hunk 1 above)
Comment (hfaerber) [reply module2/lessons/js_event_listeners.md:52]: great call!  Added

File: module2/lessons/js_event_listeners.md (hunk 2 above)
Comment (hfaerber) [reply module2/lessons/js_event_listeners.md:59]: I set up to have them switching roles for the next breakout room practice so they each get a turn in the driver seat :)

**Pull request merged**